hubtools/src/lib.rs

```diff
@@ -978,11 +1102,26 @@ mod header {
     // new-style magic numbers, which use the same header layout.
     pub(crate) const MAGIC: [u32; 2] = [0x15356637, 0x64_CE_D6_CA];
 
-    #[derive(Default, AsBytes, FromBytes)]
+    #[derive(AsBytes, FromBytes)]
     #[repr(C)]
     pub(crate) struct ImageHeader {
         pub magic: u32,
         pub total_image_len: u32,
+        pub _pad: [u32; 16],
+        pub version: u32,
+        pub epoch: u32,
+    }
+
+    impl Default for ImageHeader {
```

Comment:
It looks like this is only ever used to construct an `ImageHeader` as a function of total length and `epoch`, so I'd suggest something like
```rust
impl ImageHeader {
    fn new(total_len: usize, epoch: u32) -> Self {
        Self {
            magic: MAGIC[1],
            total_image_len: total_len as u32,
            _pad: [0; 16],
            version: 0,
            epoch,
        }
    }
}
```
Having `Default` implemented makes it easy to make an invalid header by accident if you don't overwrite the correct fields (yes, it was derived before, but was never actually used).

Reply:
Thanks